src/agent/rustjail/src/container.rs

```diff
@@ -1437,16 +1437,18 @@ impl LinuxContainer {
             Some(unistd::getuid()),
             Some(unistd::getgid()),
         )
-        .context(format!("cannot change onwer of container {} root", id))?;
+        .context(format!("Cannot change owner of container {} root", id))?;
 
         if config.spec.is_none() {
-            return Err(anyhow!(nix::Error::EINVAL));
+            return Err(anyhow!("Cannot create the container: the spec is none"));
```

Comment:
- Nit: You could make this shorter and not lose detail:

  ```suggestion
            return Err(anyhow!("Cannot create the container: no spec"));
  ```
- For strings with no format specifiers like this, I'd be tempted to define them as top-level `const` values. That also means the unit tests can reference those variables to assert the expected message when they provoke errors.

Reply:
If we want to take it a step further I could try to convert the errors to use thiserror instead of anyhow, though that might be a little overkill. I do see thiserror being used a couple other places.